projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+			UART_TX("----------------------------------------------" _EOS);
+			cmdReceived = 1; //Ensure welcome message printout
+		} else UART_TX("No Sensor Board Found! Please check connections and reset ADICUP3029."
+				       _EOS);
+		break;
+
+	case COMMAND:
+		if (check_string((uint8_t *)"PRESENCE_CHECK", 14)) {
+			if (S1_detected) UART_TX("Sensor detected on site 1" _EOS);
+			if (S2_detected) UART_TX("Sensor detected on site 2" _EOS);
+			if (S3_detected) UART_TX("Sensor detected on site 3" _EOS);
+			if (S4_detected) UART_TX("Sensor detected on site 4" _EOS);
+		}
+
+		if (check_string((uint8_t *)"READ_CFGS", 9)) {
+			if (S1_detected) GetSensorCfg(ADI_CFG_I2C_SENSOR1_ADDR);
+			delay_ms(10);
+			if (S2_detected) GetSensorCfg(ADI_CFG_I2C_SENSOR2_ADDR);
+			delay_ms(10);
+			if (S3_detected) GetSensorCfg(ADI_CFG_I2C_SENSOR3_ADDR);
+			delay_ms(10);
+			if (S4_detected) GetSensorCfg(ADI_CFG_I2C_SENSOR4_ADDR);
+			delay_ms(10);
+		}
+
+		if (check_string((uint8_t *)"READ_TEMP", 9)) {
+			readChar = GetChar();
+			DeleteChar(1);
+			eSensorResult = pGasSensor->openWithAddr(readChar);
+			flushBuff(TXbuff, sizeof(TXbuff));
+			eSensorResult = pGasSensor->ReadTemperature(&sigVal);
+			if (eSensorResult != SENSOR_ERROR_NONE) UART_TX("ERROR!" _EOS);
+			else {
+				sprintf((char*)TXbuff,"Temperature = %2.1f degC%s", (sigVal/100.0), _EOS);
+				UART_TX((const char*)TXbuff);
+			}
+			pGasSensor->close();
+		}
+
+		if (check_string((uint8_t *)"READ_HUM", 8)) {
+			readChar = GetChar();
+			DeleteChar(1);
+			eSensorResult = pGasSensor->openWithAddr(readChar);
+			flushBuff(TXbuff, sizeof(TXbuff));
+			eSensorResult = pGasSensor->ReadHumidity(&sigVal);
+			if (eSensorResult != SENSOR_ERROR_NONE) UART_TX("ERROR!" _EOS);
+			else {
+				sprintf((char*)TXbuff,"Humidity = %2.1f %%RH%s", (sigVal/100.0), _EOS);
+				UART_TX((const char*)TXbuff);
+			}
+			pGasSensor->close();
+		}
+
+		if (check_string((uint8_t *)"CFG_MEASTIME", 12)) {
+			readChar = GetChar();
+			DeleteChar(1);
+			eSensorResult = pGasSensor->openWithAddr(readChar);
+			unsigVal = (uint16_t)ExtractCfgVal();
+			flushBuff(TXbuff, sizeof(TXbuff));
+			eSensorResult = pGasSensor->SetMeasurementTime(unsigVal);
+			if (eSensorResult != SENSOR_ERROR_NONE) UART_TX("ERROR!" _EOS);
+			else {
+				sprintf((char*)TXbuff,"Measurement time set to %d msec%s", unsigVal, _EOS);
+				UART_TX((const char*)TXbuff);
+			}
+			pGasSensor->close();
+		}
+
+		if (check_string((uint8_t *)"READ_MEASTIME", 13)) {
+			readChar = GetChar();
+			DeleteChar(1);
+			eSensorResult = pGasSensor->openWithAddr(readChar);
+			flushBuff(TXbuff, sizeof(TXbuff));
+			eSensorResult = pGasSensor->ReadMeasurementTime(&unsigVal);
+			if (eSensorResult != SENSOR_ERROR_NONE) UART_TX("ERROR!" _EOS);
+			else {
+				sprintf((char*)TXbuff,"Measurement time is currently %d msec%s", unsigVal,
+					_EOS);
+				UART_TX((const char*)TXbuff);
+			}
+			pGasSensor->close();
+		}
+
+		if (check_string((uint8_t *)"STARTMEAS", 9)) {
```

Comment:
All of this if's inside the switch are a bit hard to follow. This creates a very big function with a lot of if's inside a switch.

Reply:
Understood. This also follows a similar comment by @commodo above. I will work on pulling the FSM states out into separate functions.